boards/arm/efm32wg_stk3800/Kconfig

```diff
@@ -0,0 +1,17 @@
+# Kconfig - EFM32WG STK3800 board configuration
+#
+# Copyright (c) 2017 Christian Taedcke
+#
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_EFM32WG_STK3800
+
+config BOARD_INIT_PRIORITY
```

Comment:
any particular reason to make this configurable?

Reply:
This it not necessary, but it must be executed after the initialization of the gpio driver, which has the init priority CONFIG_KERNEL_INIT_PRIORITY_DEFAULT
How should i do this in a better way?

Reply:
Just hard code the value, my point was more about it being configurable.

something like:

/* needs to be done after GPIO driver init, so +1 on Kernel init prio */
SYS_INIT(efm32wg_stk3800_init, PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT+1)

Reply:
I remember that, back in the days when we had issues with level/prio ordering that it got decided not do this trick of CONIFG_SOME_PRIO + extra value. Better have you own Kconfig option, you can make it hidden one as long as there is no prompt.

Reply:
do you remember why that was decided, it seems really odd to make this a Kconfig thing, seems pretty error prone (vs the +1)

Reply:
I'll try to see if the thread is still in my mail box

Reply:
This does not compile:

~~~
SYS_INIT(efm32wg_stk3800_init, PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT+1)
~~~

~~~
  CC      boards/arm/efm32wg_stk3800/board.o
In file included from /home/chrta/projects/projects/zephyr/include/init.h:11:0,
                 from /home/chrta/projects/projects/zephyr/boards/arm/efm32wg_stk3800/board.c:7:
/home/chrta/projects/projects/zephyr/include/device.h:107:16: error: expected ‘,’ or ‘;’ at end of input
  static struct device _CONCAT(__device_, dev_name) __used \
                ^
/home/chrta/projects/projects/zephyr/include/device.h:159:2: note: in expansion of macro ‘DEVICE_AND_API_INIT’
  DEVICE_AND_API_INIT(dev_name, drv_name, init_fn, data, cfg_info, \
  ^~~~~~~~~~~~~~~~~~~
/home/chrta/projects/projects/zephyr/include/init.h:50:2: note: in expansion of macro ‘DEVICE_INIT’
  DEVICE_INIT(_SYS_NAME(init_fn), "", init_fn, NULL, NULL, level, prio)
  ^~~~~~~~~~~
/home/chrta/projects/projects/zephyr/boards/arm/efm32wg_stk3800/board.c:34:1: note: in expansion of macro ‘SYS_INIT’
 SYS_INIT(efm32wg_stk3800_init, PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT+1)
 ^~~~~~~~

~~~

I suspect it is because of this line in include/device.h

~~~
 __attribute__((__section__(".init_" #level STRINGIFY(prio)))) = { \
~~~

So the prio must play nice with STRINGIFY()

Reply:
The calculation of the init priority does not play well with STRINGIFY:
~~~
/opt/zephyr-sdk/sysroots/x86_64-pokysdk-linux/usr/libexec/arm-zephyr-eabi/gcc/arm-zephyr-eabi/6.2.0/real-ld: Undefined initialization levels used.
~~~

~~~
*(SORT(.init_[_A-Z0-9]*))
 .init_PRE_KERNEL_140+1
                0x0000000020001018        0xc boards/built-in.o
                0x0000000000000000                ASSERT ((SIZEOF (initlevel_error) == 0x0), Undefined initialization levels used.)


~~~

Reply:
Ok, I'm fine with leaving this as a hidden config option for now.

Reply:
I made this Kconfig value hidden and added comments about its relation to the gpio init priority.